doc/configuration/general.md

```diff
@@ -190,24 +190,3 @@ Replaces [Cowboy](https://github.com/ninenines/cowboy)'s default name returned i
 
 When specified, will terminate certain processes (e.g. client handlers) that have more messages accumulated in the queue than the specified limit, to prevent resource exhaustion.
 This option is set for outgoing S2S connections and can be overridden for particular `s2s` listeners in their configurations. **Use with caution!**
-
-### `general.domain_certfile`
```

Comment:
This is quite a useful option, why do we remove it?

Reply:
Because now we configure certificates using the same tls section-style than c2s and components have, all listeners now use the same tls engine in the same way so config is also the same:

https://github.com/esl/MongooseIM/blob/06a7d3a7c119a0113d1b6ae0a8c0a37d1cef7c52/src/config/mongoose_config_spec.erl#L317

https://github.com/esl/MongooseIM/blob/06a7d3a7c119a0113d1b6ae0a8c0a37d1cef7c52/src/config/mongoose_config_spec.erl#L627

Reply:
this is not equivalent to the `general.domain_certfile`

Reply:
Nope, but close enough.
- Overriding the domains from the TLS config with these global ones makes code much uglier.
- This was used only by S2S, not by C2S nor connection, already since many years.
- Also, in the case of direct TLS connections, you don't know the server before having to establish the TLS connection, so you can't override the certificate.

Reply:
as discussed during the call, an alternative to `general.domain_certfile` is creation of a multiple listeners with DNS [SRV](https://en.wikipedia.org/wiki/SRV_record) records pointing to the proper port. please add this information to the migration guidelines in the upcoming PRs.